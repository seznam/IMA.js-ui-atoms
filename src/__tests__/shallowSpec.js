import classnames from 'classnames';
import React from 'react';
import { shallow } from 'enzyme';
import { Infinite } from 'infinite-circle';
import { toMockedInstance } from 'to-mock';
import { withContext } from 'shallow-with-context';

import {
  Sizer,
  UIComponentHelper,
  Visibility,
  ComponentPositions,
} from '../main';

import dummyRouter from '../mocks/router';
import dummyWindow from '../mocks/window';

describe('UIAtoms shallow rendering', () => {
  let wrapper = null;
  const visibility = toMockedInstance(Visibility);
  const componentPositions = toMockedInstance(ComponentPositions);
  const infinite = toMockedInstance(Infinite);
  const uiComponentHelper = new UIComponentHelper(
    dummyRouter,
    dummyWindow,
    visibility,
    componentPositions,
    infinite,
    classnames
  );
  const context = {
    $Utils: {
      $UIComponentHelper: uiComponentHelper,
    },
  };

  afterEach(() => {
    if (wrapper) {
      wrapper.unmount();
    }
  });

  describe('Sizer component', () => {
    beforeEach(() => {
      const Component = withContext(Sizer, context);

      wrapper = shallow(<Component />, { context });
    });

    it('should render sizer and set atm-sizer class', () => {
      expect(wrapper.html()).toMatchSnapshot();
      expect(wrapper.hasClass('atm-sizer')).toBeTruthy();
    });

    it('should set atm-placeholder class if is defined placeholder props', () => {
      wrapper.setProps({ placeholder: true });

      expect(wrapper.hasClass('atm-placeholder')).toBeTruthy();
    });

    it('should calculate ratio between width and height', () => {
      wrapper.setProps({
        width: 16,
        height: 9,
      });

      expect(wrapper.html()).toMatchSnapshot();
      expect(wrapper.get(0).props.style.paddingTop).toEqual('56.25%');
    });
  });
});
