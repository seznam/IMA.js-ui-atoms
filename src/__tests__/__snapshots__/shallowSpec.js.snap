// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UIAtoms shallow rendering Sizer component should render sizer and set atm-sizer class 1`] = `"<div class=\\"atm-sizer\\" style=\\"padding-top:NaN%\\"></div>"`;
